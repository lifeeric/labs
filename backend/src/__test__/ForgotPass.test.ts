import * as util from "util";
import axios from "axios";
import { exec } from "child_process";
import path from "path";

// const pExec = util.promisify(exec);

const API = "http://localhost:8000/graphql";
// const CMD_SEED_DB = path.join(__dirname, "/../../dataset.sh");

// Invoke axios
const axios_post = async (API: string, query: string) => {
  return await axios.post(API, {
    query,
  });
};

describe("ForgotPass testing", () => {
  // beforeAll(async () => {
  //   await pExec(CMD_SEED_DB);
  // });

  it("should get the data", async () => {
    const query = `
    query {
        forgotPassword(email: "alex@test.com") {
         __typename
        ... on ForgetPassordResult {
          id
          email
        }
        
        ... on Error {
          error
          message
        }
      }
    }`;

    const response = await axios_post(API, query);
    expect(response.data).toMatchSnapshot();
  });
});
