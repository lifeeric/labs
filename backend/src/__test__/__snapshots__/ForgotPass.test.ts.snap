// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPass testing should get the data 1`] = `
Object {
  "data": Object {
    "forgotPassword": Object {
      "__typename": "ForgetPassordResult",
      "email": "alex@test.com",
      "id": "5f449b73e2ccbc43aa520488",
    },
  },
}
`;
